Framework/PythonInterface/mantid/plots/plotfunctions.py

```diff
@@ -357,9 +359,176 @@ def pcolormesh(axes, workspace, *args, **kwargs):
 
     return axes.pcolormesh(x, y, z, *args, **kwargs)
 
+def _imshow(axes, X, cmap=None, norm=None, aspect=None,
```

Comment:
Please rename the values being plotted to be `z` rather than `X`. I think that will be less confusing. Also, you should reduce the parameters to be `_imshow(axes, z, * args, **kwargs)` then use code like `norm = kwargs.pop('norm', None)` below.